src/common/hist_util.cu

```diff
@@ -82,6 +84,18 @@ __global__ void UnpackFeaturesK
   }
 }
 
+/*!
+ * \brief A container that holds the device sketches across all
+ *  sparse page batches which are distributed to different devices.
+ *  As sketches are aggregated by column, the mutex guards
+ *  multiple devices pushing sketch summary for the same column
+ *  across distinct rows.
+ */
+struct SketchContainer {
+  std::vector<HistCutMatrix::WXQSketch> sketches_;  // NOLINT
+  std::vector<std::unique_ptr<std::mutex>> col_locks_; // NOLINT
```

Comment:
Feel free to use `std::vector<std::mutex>`, unless there is a reason to use `std::unique_ptr`.

Reply:
elements within a vector have to be copy/assign-able or move-able. a mutex is neither of those.

Reply:
I think copying/assignment/moving only applies when resizing the vector. Creating a mutex vector of a fixed size should be possible.

I'm not a C++ expert here, however, so I may be wrong.